.github/workflows/ci.yml

```diff
@@ -0,0 +1,191 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+jobs:
+  lint:
+    name: Lint
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 1
+    - uses: actions/setup-node@v2-beta
+      with:
+        node-version: '10.x'
+    - name: Get package manager's global cache path
+      id: global-cache-dir-path
+      run: echo "::set-output name=dir::$(npm config get cache)"
+    - name: Cache package manager's global cache
+      id: cache-global-package-manager-cache
+      uses: actions/cache@v1
+      with:
+        path: ${{ steps.global-cache-dir-path.outputs.dir }}
+        key: ${{ runner.os }}-${{ matrix.node-version }}-yarn-${{ hashFiles('**/package-lock.json') }}
+        restore-keys: |
+          ${{ runner.os }}-${{ matrix.node-version }}-yarn-
+    - name: Cache node_modules
+      id: cache-node-modules
+      uses: actions/cache@v1
+      with:
+        path: node_modules
+        key: ${{ runner.os }}-${{ matrix.node-version }}-nodemodules-${{ hashFiles('**/package-lock.json') }}
+        restore-keys: |
+          ${{ runner.os }}-${{ matrix.node-version }}-nodemodules-
+    - name: Install Dependencies
+      run: npm ci
+      if: |
+        steps.cache-global-package-manager-cache.outputs.cache-hit != 'true' ||
+        steps.cache-node-modules.outputs.cache-hit != 'true'
```

Comment:
Suggestions

- Use `actions/cache@v2` to simplify steps
- For `npm`, be careful with caching if the OS includes Windows, not just Ubuntu/Mac

https://gist.github.com/ijlee2/fba20304670b8d2b71f0b81cd07ea026#file-ci-yml-L71-L83

Reply:
The key (line 28) includes the word `yarn`. In the updated workflow template (linked above), I removed the word `yarn`/`npm` for convenience.